packages/ui/src/components/va-avatar/VaAvatar.vue

```diff
@@ -24,47 +23,51 @@
 </template>
 
 <script lang="ts">
-import { Options, mixins, prop, Vue } from 'vue-class-component'
+import { defineComponent, PropType, computed } from 'vue'
+
+import { useColors } from '../../services/color-config/color-config'
+import { useSize, useSizeProps } from '../../composables/useSize'
+import { useLoadingProps } from '../../composables/useLoading'
 
-import ColorMixin from '../../services/color-config/ColorMixin'
-import { SizeMixin } from '../../mixins/SizeMixin'
-import { LoadingMixin } from '../../mixins/LoadingMixin/LoadingMixin'
 import VaIcon from '../va-icon'
 import { VaProgressCircle } from '../va-progress-bar'
 
-class AvatarProps {
-  color = prop<string>({ type: String, default: 'info' })
-  textColor = prop<string>({ type: String, default: 'white' })
-  square = prop<boolean>({ type: Boolean, default: false })
-  icon = prop<string>({ type: String, default: '' })
-  src = prop<string>({ type: String, default: null })
-  fontSize = prop<string>({ type: String, default: '' })
-}
-
-const AvatarPropsMixin = Vue.with(AvatarProps)
-
-@Options({
+export default defineComponent({
   name: 'VaAvatar',
   components: { VaIcon, VaProgressCircle },
-})
-export default class VaAvatar extends mixins(
-  SizeMixin,
-  ColorMixin,
-  LoadingMixin,
-  AvatarPropsMixin,
-) {
-  get computedStyle () {
+  props: {
+    ...useLoadingProps,
+    ...useSizeProps,
+    color: { type: String as PropType<string>, default: 'info' },
+    textColor: { type: String as PropType<string>, default: 'white' },
+    square: { type: Boolean as PropType<boolean>, default: false },
+    icon: { type: String as PropType<string>, default: '' },
+    src: { type: String as PropType<string>, default: null },
+    fontSize: { type: String as PropType<string>, default: '' },
+  },
+  setup (props) {
+    const { getColor } = useColors()
+    const { sizeComputed, fontSizeComputed } = useSize(props, 'VaAvatar')
+
+    const colorComputed = computed(() => getColor(props.color))
+
+    const computedStyle = computed(() => ({
+      color: getColor(props.textColor, '#ffffff'),
```

Comment:
```suggestion
      color: getColor(props.textColor, 'white'),
```

Use white from colors config. For dark theme we can set white to `#000000` and VaAvatar should depend on this color.